src/app/drive/components/DriveExplorer/DriveExplorerOverlay/DriveExplorerOverlay.tsx

```diff
@@ -1,21 +1,45 @@
-import './DriveExplorerOverlay.scss';
+import { Upload } from 'phosphor-react';
+import { ReactNode } from 'react';
 
 interface DriveExplorerOverlayProps {
   icon: JSX.Element;
   title: string;
   subtitle: string;
+  action?: {
+    text: string;
+    icon: typeof Upload;
+    style: 'plain' | 'elevated';
+    onClick: () => void;
+  };
 }
 
-function DriveExplorerOverlay({ icon, title, subtitle }: DriveExplorerOverlayProps): JSX.Element {
+function DriveExplorerOverlay({ icon, title, subtitle, action }: DriveExplorerOverlayProps): JSX.Element {
+  let button: ReactNode = null;
+
+  if (action) {
+    button = (
+      <button
+        onClick={action.onClick}
+        className={`mx-auto flex items-center rounded-lg ${
+          action.style === 'elevated' ? 'mt-5 bg-primary' : 'mt-2.5 bg-transparent'
+        }  bg-opacity-10 px-6 py-2.5 font-medium text-primary hover:bg-opacity-15 active:bg-opacity-20`}
```

Comment:
"px-6 py-2.5" should be "px-5 h-10" (padding-x of 20px and fixed height at 40px)